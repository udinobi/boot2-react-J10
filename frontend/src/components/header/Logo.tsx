import React from "react"
import styled, { keyframes } from "styled-components"

import logo from "../../assets/logo.png"

const rotate360 = keyframes`
    from {
        transform: rotate(0deg);
    }
    to {
        transform: rotate(360deg);
    }
`

const Logo = styled.img`
    animation: ${rotate360} infinite 120s linear;
    height: 80px;
    &:hover {
        animation: ${rotate360} infinite 1.5s linear;
    }
    z-index: 10;
    margin-left: 12px;
    margin-top: 12px;
`

export default () =>
    <Logo src={logo} alt="'Weather In The World' logo"/>
