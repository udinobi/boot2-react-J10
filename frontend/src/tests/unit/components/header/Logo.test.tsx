import React from "react"
import * as Renderer from 'react-test-renderer'

import Logo from "../../../../components/header/Logo"

describe("--> Snapshot for Components > header > 'Logo'", () => {
    it("matches", () => {
        const component = Renderer.create(<Logo />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
