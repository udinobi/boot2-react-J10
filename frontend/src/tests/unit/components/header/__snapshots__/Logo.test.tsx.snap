// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > header > 'Logo' matches 1`] = `
<img
  alt="'Weather In The World' logo"
  className="sc-bdVaJa fQqfno"
  src="logo.png"
/>
`;
